src/instantiate.js

```diff
@@ -0,0 +1,584 @@
+        if (curModule.indexOf('*') != -1) {
+          var parts = curModule.split('*');
+          if (parts.length != 2) {
+            config.bundles[b].splice(i--, 1);
+            continue;
+          }
+
+          if (key.substring(0, parts[0].length) == parts[0] &&
+              key.substr(key.length - parts[1].length, parts[1].length) == parts[1] &&
+              key.substr(parts[0].length, key.length - parts[1].length - parts[0].length).indexOf('/') == -1) {
+            matched = true;
+            break;
+          }
+        }
+      }
+
+      if (matched)
+        return loader.import(b);
+    }
+  }
+}
+
+function runFetchPipeline (loader, key, metadata, processAnonRegister, wasm) {
+  if (metadata.load.exports && !metadata.load.format)
+    metadata.load.format = 'global';
+
+  return Promise.resolve()
+
+  // locate
+  .then(function () {
+    if (!metadata.pluginModule || !metadata.pluginModule.locate)
+      return;
+
+    return metadata.pluginModule.locate.call(loader, metadata.pluginLoad)
+    .then(function (address) {
+      if (address)
+        metadata.pluginLoad.address = address;
+    });
+  })
+
+  // fetch
+  .then(function () {
+    if (!metadata.pluginModule)
+      return fetch(key, metadata.load.authorization, metadata.load.integrity, wasm);
+
+    if (!metadata.pluginModule.fetch)
+      return fetch(metadata.pluginArgument, metadata.load.authorization, metadata.load.integrity, wasm);
+
+    wasm = false;
+    return metadata.pluginModule.fetch.call(loader, metadata.pluginLoad, function (load) {
+      return fetch(load.address, metadata.load.authorization, metadata.load.integrity, false);
+    });
+  })
+
+  .then(function (fetched) {
+    // fetch is already a utf-8 string if not doing wasm detection
+    if (!wasm)
+      return translateAndInstantiate(loader, key, fetched, metadata, processAnonRegister);
+
+    var bytes = new Uint8Array(fetched);
+
+    // detect by leading bytes
+    if (bytes[0] === 0 && bytes[1] === 97 && bytes[2] === 115) {
+      return WebAssembly.compile(bytes).then(function (m) {
+        /* TODO handle imports when `WebAssembly.Module.imports` is implemented
+        if (WebAssembly.Module.imports) {
+          var deps = [];
+          var setters = [];
+          var importObj = {};
+          WebAssembly.Module.imports(m).forEach(function (i) {
+            var key = i.module;
+            setters.push(function (m) {
+              importObj[key] = m;
+            });
+            if (deps.indexOf(key) === -1)
+              deps.push(key);
+          });
+          loader.register(deps, function (_export) {
+            return {
+              setters: setters,
+              execute: function () {
+                _export(new WebAssembly.Instance(m, importObj).exports);
+              }
+            };
+          });
+        }*/
+        // for now we just load WASM without dependencies
+        var exports = new WebAssembly.Instance(m, {}).exports;
```

Comment:
nvm, misread the code. (Can you split the line into two for better reading? `var wasmModule = new WebAssembly.Instance(m, {});` and `var exports = wasmModule.exports;` ?)